tests/tools/fio/workload.go

```diff
@@ -177,24 +188,26 @@ func (fr *Runner) writeFilesAtDepth(fromDirPath string, depth, branchDepth int,
 }
 
 func pickRandSubdirPath(dirPath string) (subdirPath string) {
-	subdirCount := 0
-
 	fileInfoList, err := ioutil.ReadDir(dirPath)
 	if err != nil {
 		return ""
 	}
 
-	for _, fi := range fileInfoList {
+	// Find all enties that are directories, record each of their fileInfoList indexes
+	dirIdxs := make([]int, 0, len(fileInfoList))
+	for idx, fi := range fileInfoList {
```

Comment:
This works. The previous approach works as well, just adding a comment saying it's doing reservoir sampling should be enough for the clueless reviewers 😉